DuckDuckGo/IPC/VPNCommander.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  VPNCommander.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+struct VPNCommander {
+
+    func restartAdapter() async throws {
+        guard let activeSession = await AppDependencyProvider.shared.networkProtectionTunnelController.activeSession(),
+            activeSession.status == .connected else {
+
+            return
+        }
+
+        try? await activeSession.sendProviderRequest(.command(.restartAdapter))
```

Comment:
What do you think about instead putting this in `NetworkProtectionTunnelController`? (I wonder whether the tunnel controller should have a function like `sendProviderRequest` that abstracts the fetching of the session etc., since we have similar logic in a few places.)

Reply:
Agreed.  I was avoiding it because this forces the change on macOS (as I need to modify the `TunnelController` protocol) which requires some thinking.

That said, I think it makes sense so I pushed the change.